ministryofjustice/laa-review-criminal-legal-aid
# [CRIMRE-212] Restrict user deactivation journey

## Description of change
Ensure that the user cannot delete all other users in manage_users dashboard or edit user 'can_manage_others' permission without there being 2 admins remaining.

## Link to relevant ticket
https://dsdmoj.atlassian.net/browse/CRIMRE-212

## Notes for reviewer
Ensures there are always 2 admins (i.e. a user with `can_manage_others` flag set to `true`.

![Screenshot 2023-06-07 at 10 26 34](https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/47113046/7de2dc62-6474-4e5f-a6f3-71c1cbc48eb0)

![Screenshot 2023-06-07 at 10 29 41](https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/47113046/00771dd5-9fc6-4f47-8b00-2e594ea7b1e3)



## Screenshots of changes (if applicable)

### Before changes:

### After changes:

#### Attempting to deactivate the last admin:
![image](https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/106679087/76c098d6-1985-49e2-8ce5-58d292ab8233)

#### Attempting to edit the last admin:
![Screenshot 2023-06-07 at 09 52 20](https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/47113046/0c8b1675-b1f1-4e41-bb7b-f03e1f6ad390)


## How to manually test the feature
- User logs into portal.
- If the user has manage other user's responsibility they should see an option to manage users as attached on the screenshot.
- User clicks on Manage users.
- if the user has existing users in the manage users menu, they can edit the users and give them access to manage other users permissions, by clicking on edit under the Actions subheading.
- If the user has multiple existing users they can deactivate the users by clicking on deactivate under the Actions subheading.
- When the user can see that there only two admin users left and they try to deactivate another admin user they will get message banner that will say "Unable to deactivate user. There must be at least two users who can manage others".
